src/compiler/checker.ts

```diff
@@ -41357,12 +41378,15 @@ namespace ts {
                 if (flags & ModifierFlags.Override) {
                     return grammarErrorOnNode(lastOverride!, Diagnostics._0_modifier_cannot_appear_on_a_constructor_declaration, "override"); // TODO: GH#18217
                 }
-                else if (flags & ModifierFlags.Async) {
+                if (flags & ModifierFlags.Async) {
                     return grammarErrorOnNode(lastAsync!, Diagnostics._0_modifier_cannot_appear_on_a_constructor_declaration, "async");
                 }
-                else if (flags & ModifierFlags.Readonly) {
+                if (flags & ModifierFlags.Readonly) {
                     return grammarErrorOnNode(lastReadonly!, Diagnostics._0_modifier_cannot_appear_on_a_constructor_declaration, "readonly");
                 }

```

Comment:
This seems like it deserves to move to its own PR

Reply:
@DanielRosenwasser I removed the `noise` changes